### ISSUE TITLE
Fix false positives for objects in declaration-block-trailing-semicolon

### ISSUE THREAD

**Issue by dreyks** [stylelint/stylelint#4650]
> Clearly describe the bug

Stylelint emits a warning when declaration-block-trailing-semicolon meets styles in jsx

> Which rule, if any, is the bug related to?

`declaration-block-trailing-semicolon`

> What code is needed to reproduce the bug?

```js
const Component = () => {
  const background = 'white'

  return <div style={{ background }}>Text</div>
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "declaration-block-trailing-semicolon": "always"
  }
}
```

> Which version of stylelint are you using?

e.g. `13.2.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to JSX

> What did you expect to happen?

no warnings

> What actually happened (e.g. what warnings or errors did you get)?

```
Expected a trailing semicolon   declaration-block-trailing-semicolon
```

I've monkeypatched this in my code through patch-package by adding 

```diff
diff --git a/node_modules/stylelint/lib/rules/declaration-block-trailing-semicolon/index.js b/node_modules/stylelint/lib/rules/declaration-block-trailing-semicolon/index.js
index 890dd96..052e43c 100644
--- a/node_modules/stylelint/lib/rules/declaration-block-trailing-semicolon/index.js
+++ b/node_modules/stylelint/lib/rules/declaration-block-trailing-semicolon/index.js
@@ -40,6 +40,10 @@ function rule(expectation, _, context) {
 		});
 
 		root.walkDecls((decl) => {
+			if (decl.parent.type === 'object') {
+				return;
+			}
+
 			if (decl !== decl.parent.last) {
 				return;
 			}
```

**Comment by jeddy3** [comment]
@dreyks Thanks for the report, digging into a fix and for using the template.

We can improve and use the [`isStandardSyntaxRule` utility](https://github.com/stylelint/stylelint/blob/master/lib/utils/isStandardSyntaxRule.js).

```js
// utils/isStandardSyntaxRule.js
if (rule.type !== 'rule') return false;
```

```js
// declaration-block-trailing-semicolon/index.js
if (!isStandardSyntaxRule(decl.parent)) return;
```

This will benefit other similar rules too.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.
